cvat/apps/dataset_manager/formats/utils.py

```diff
@@ -62,18 +63,43 @@ def make_colormap(instance_data):
     return {label['name']: [hex2rgb(label['color']), [], []] for label in labels}
 
 
-def get_label_color(label_name, label_names):
+
+def generate_color(color, used_colors):
+    def tint_shade_color():
+        new_color = [0, 0, 0]
+        for added_color in (255, 0):
+            for factor in range(1, 10):
+                for c in range(0, 3):
+                    new_color[c] = int(color[c] + (added_color - color[c])*factor/10)
+                yield tuple(new_color)
+
+    def get_unused_color():
```

Comment:
I would recommend another algorithm to get unused colors:
- Sort all used_colors by x, y, z -> [(0, 0, 0), (10, 30, 22), (255, 0, 0)]
- Find 2 adjacent points with the biggest difference by one of coordinates: (10, 30, 22), (255, 0, 0). Because dx = 255 - 10 is the maximum and they are adjacent.
- Add a new point as ((x0 + x1) // 2, y0, z0). In our case (132, 30, 22).